### ISSUE TITLE
Run RevAPI without Gradle

### ISSUE THREAD

**Issue by Fokko** [apache/iceberg#10368]
### Feature Request / Improvement

By decoupling RevAPI from Gradle we can upgrade to a later version of Gradle.

A later version of Gradle is required to unblock #10209. The 1.14.0 version of Parquet ships with a newer version of Jackson, with JDK21-specific files. These files cause issues with the older version of the Shadow plugin that we're using. The shadow plugin requires a later version of Gradle.

We must double-check if RevAPI is working properly since it silently stopped working when updating Gradle to a later version.

Blocks #8485

I believe @jbonofre wants to work on this, but he can also work on it if anyone else is interested.

### Query engine

None

**Comment by jbonofre** [comment]
Yes I have something about that coupled with gradle update. I will push the PR asap.

**Comment by jbonofre** [comment]
I created a script to run revapi: https://github.com/jbonofre/iceberg/blob/GRADLE/dev/revapi

I have three things to address:
1. support `acceptedBreaks` from `.palantir/revapi.yml` (I might need more than a script for that), I'm checking different options
2. the script should failed in error when revapi detects a breaking change
3. integrate the script in gradle

**Comment by jbonofre** [comment]
After some investigations, I'm doing:
1. I'm replacing `.palantir/revapi.yml` (specific to the revapi gradle plugin) by a simple file where users can add accepted breaks
2. the script now detects a breaking change by grepping the output report
3. I'm not sure if the script should be integrated in gradle or not. Probably worth to execute the script after the build (the script needs to be run after `compileJava` task).

**Comment by findepi** [comment]
How does revapi manifests that it does not support newer gradle?

**Comment by findepi** [comment]
@jbonofre do we already know the root cause why the rev api stops working when we upgrade grade? If it something not our side, are the plug-in authors aware of the problem?

**Comment by ajantha-bhat** [comment]
@findepi: The problem is gradle plugin is not officially part of rev API maintainers.  
It is a separate group and they stopped maintaining it (last release was 2022). 

We also created the issue but no one responded
https://github.com/palantir/gradle-revapi/issues/612

**Comment by jbonofre** [comment]
@findepi yes I have the fix for the plugin. But my concern is the plugin is not maintained anymore (no commit or release in the last 2 years).

**Comment by ajantha-bhat** [comment]
I started this discussion also: https://github.com/revapi/revapi/issues/296

**Comment by Fokko** [comment]
Not needed anymore https://github.com/apache/iceberg/pull/8486
